src/components/PatchTabs/downstreamTasks/reducer.ts

```diff
@@ -0,0 +1,141 @@
+import { SortOrder } from "gql/generated/types";
+
+interface QueryParamState {
+  baseStatuses: string[];
+  limit: number;
+  page: number;
+  sorts: SortOrder[];
+  statuses: string[];
+  taskName: string;
+  variant: string;
+}
+
+interface InputValueState {
+  baseStatusesInputVal: string[];
+  currentStatusesInputVal: string[];
+  taskNameInputVal: string;
+  variantInputVal: string;
+}
+
+export type Action =
+  | { type: "onChangeBaseStatusesSelector"; data: string[] }
```

Comment:
nit. instead of a generic `data` parameter we should prob include a more accurate name of the field that we are passing in. So the caller knows what kind of information should be passed in.